### PR TITLE
Censor credentials in debug log messages.

### PR DESCRIPTION
**Pull request by Smartkras**
Censoring of credentials was added to debug log messages for Windows, Android and curl implementation.

Resolves: OAM-2216

### CODE REVIEW

File: olp-cpp-sdk-core/include/olp/core/utils/Credentials.h [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 2019-2023 HERE Europe B.V.
```
Comment (andrey-kashcheev) [inline olp-cpp-sdk-core/include/olp/core/utils/Credentials.h:2]: Here and in the cpp 2023 only, as this is a new file technically

File: olp-cpp-sdk-core/tests/utils/UtilsTest.cpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+#include <string>
+
+#include <olp/core/utils/Credentials.h>
+
+namespace {
+
+using UtilsTest = testing::Test;
+
+TEST(UtilsTest, Credentials) {
```
Comment (andrey-kashcheev) [inline olp-cpp-sdk-core/tests/utils/UtilsTest.cpp:30]: Maybe split with SCOPED_TRACE? 
- Empty url
- Nothing to censor
- Censoring app_id, app_code

File: olp-cpp-sdk-core/src/utils/Credentials.cpp [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include <olp/core/utils/Credentials.h>
+
+namespace {
+void censor_credentials_part(std::string& url, std::size_t arguments_start,
```
Comment (mykhailo-kuchma) [inline olp-cpp-sdk-core/src/utils/Credentials.cpp:23]: CensorCredentialsPart

File: olp-cpp-sdk-core/src/utils/Credentials.cpp [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  const auto arguments_start_pos = url.find('?');
+
+  if (arguments_start_pos == std::string::npos) {
+    return url;
+  }
+
+  static const std::string app_id_name = "app_id=";
+  static const std::string app_code_name = "app_code=";
+  censor_credentials_part(url, arguments_start_pos, app_id_name);
+  censor_credentials_part(url, arguments_start_pos, app_code_name);
```
Comment (mykhailo-kuchma) [inline olp-cpp-sdk-core/src/utils/Credentials.cpp:62]: Can you also try the API key authentication and censor it

**Review by mykhailo-kuchma — CHANGES_REQUESTED** [review]

**Review by andrey-kashcheev — APPROVED** [review]

**Review by mykhailo-kuchma — APPROVED** [review]

**Pull request merged**